libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/search.hpp

```diff
@@ -9,19 +10,281 @@
 #pragma once
 
 #include <hpx/config.hpp>
+#include <hpx/execution/algorithms/detail/predicates.hpp>
 #include <hpx/execution/traits/is_execution_policy.hpp>
 #include <hpx/iterator_support/range.hpp>
 #include <hpx/iterator_support/traits/is_range.hpp>
 
 #include <hpx/algorithms/traits/projected.hpp>
 #include <hpx/algorithms/traits/projected_range.hpp>
+// Still need to include the non-range header since we don't have search_n
+// sequential implementation supporting sentinels yet
```

Comment:
Is this comment relevant? I think we include the non-range header for *all* the adapted algorithms. If anything we could start moving the implementations into something like `hpx/parallel/algorithms/detail`, but at the moment we always include the non-range header because it contains the actual algorithm implementations.

Reply:
When I used the range-v3 sequential impl, it made sense to put the detail implementation in another file because of the copyright. And I planned to adapt the search_n algorithm (other PR) in the same file and the container algos would then only depend on the detail file. It made sense for me as some other algos also have a detail file. But if you prefer having the detail impl in the main `algorithms/search.hpp` file I can move them back

Reply:
I think `hpx/parallel/algorithms/detail/search.hpp` might make sense. We currently have the sequential, sentinel-supporting versions of algorithms in there.

Reply:
Could I suggest moving the `search_n` implementation to the detail header already now even if it doesn't support sentinels? It's seems to me that sentinel support is an orthogonal concern to having to include the non-range header.